connectors/katalog/pkg/connector/connector.go

```diff
@@ -95,30 +100,36 @@ func (r *Handler) createAsset(c *gin.Context) {
 		return
 	}
 
+	errString := "Error during create asset! Error:"
+	secretName, secretNamespace, err := vault.GetKubeSecretDetailsFromVaultPath(request.Credentials)
+	if err != nil {
+		r.reportError(c, http.StatusInternalServerError, errString+err.Error())
+		return
+	}
+
 	asset := &v1alpha1.Asset{
 		ObjectMeta: v1.ObjectMeta{Namespace: request.DestinationCatalogID, Name: request.DestinationAssetID, GenerateName: FybrikAssetPrefix},
 		Spec: v1alpha1.AssetSpec{
-			SecretRef: v1alpha1.SecretRef{Name: request.Credentials},
+			SecretRef: v1alpha1.SecretRef{Name: secretName, Namespace: secretNamespace},
 			Metadata:  request.ResourceMetadata,
 			Details:   request.Details,
 		},
 	}
 
 	logging.LogStructure("Fybrik Asset to be created in Katalog:", asset, &r.log, zerolog.DebugLevel, false, false)
 
-	err := r.client.Create(context.Background(), asset)
+	err = r.client.Create(context.Background(), asset)
 	if err != nil {
-		errString := "Error during create asset!"
 		if errors.IsAlreadyExists(err) {
-			errString = "Asset Already exists!"
+			errString = "Asset Already exists!, Error:"
```

Comment:
no need for prefix

Reply:
So the asset name is already taken. Use it as a prefix, returning an error may be inconsistent with the behavior of other connectors.

Reply:
As discussed assetId is used as prefix if exists.